### PR TITLE
Nit: allocate slice capacity correctly during intersection

### PR DESCRIPTION
**Pull request by yeya24**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

A very small nit.

When both posting groups have positive matchers (with add keys), we perform an intersection operation.
For intersection two slices a and b, the final result size can be at maximum `min(len(a), len(b))`. Allocating `len(pg.addKeys)+len(other.addKeys)` size is unnecessary.

### CODE REVIEW

File: pkg/store/bucket.go [hunk 1]
```diff
@@ -2625,6 +2625,13 @@ func (pg postingGroup) mergeKeys(other *postingGroup) *postingGroup {
 	return &pg
 }
 
+func min(a, b int) int {
```
Comment (fpetkovski) [inline pkg/store/bucket.go:2628]: I think we have these functions in go 1.21, so maybe we can just use them?

Comment (yeya24) [reply pkg/store/bucket.go:2628]: Good point. I will take a look at how to do it.

Comment (fpetkovski) [reply pkg/store/bucket.go:2628]: I think it's fine to do it, eventually we'll want to use features from new Go versions.

Comment (yeya24) [reply pkg/store/bucket.go]: @fpetkovski Updated! PTAL

**Review by fpetkovski — APPROVED** [review]

**Pull request merged**